### PR TITLE
Fixes memory-leaks

### LINKED ISSUE #102: Possible memory-leak in bindMethods
First, I'm not sure if this issue is limited to bindMethods, but that's what shows up in my profile, so I'm reporting it that way.

I observed that a huge number of sets were being created by emittery and never being cleaned up. Upon investigating, it looks like these sets are being created to store the various event-handler functions for event names but these sets are never destroyed and the map keys never cleared out when all of the handlers are removed.

For example, here's a snapshot of my memory profile:
image

### PR DESCRIPTION
**Pull request by gurupras**
Should fix #102

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ function assertListener(listener) {
 function getListeners(instance, eventName) {
 	const events = eventsMap.get(instance);
 	if (!events.has(eventName)) {
-		events.set(eventName, new Set());
+		return null;
```
Comment (sindresorhus) [inline index.js:31]:
```suggestion
		return;
```

I prefer not using `null`.

**Comment by sindresorhus** [comment]
Thanks for working on this. Can you add some tests?

**Comment by gurupras** [comment]
That would require some design decisions that I'm not sure I'm well-equipped to make, given that I only started poking at the codebase < 12 hours ago.

For example, currently, the maps (`eventsMap`, `producersMap`) are private and would require some refactoring to get to them.

**Comment by gurupras** [comment]
I added a test for `off()`. I don't really understand what the `producersMap` is for or how to go about testing that map, sorry.

**Pull request merged**